src/services/item/item.repository.ts

```diff
@@ -292,12 +294,32 @@ export class ItemRepository {
   /**
    * return children of given parent, ordered by order
    * @param db
+   * @param parent
+   * @returns
+   */
+  async getChildren(dbConnection: DBConnection, parent: ItemRaw): Promise<ItemRaw[]> {
+    if (parent.type !== ItemType.FOLDER) {
+      throw new ItemNotFolder({ id: parent.id });
+    }
+
+    return await dbConnection
+      .select()
+      .from(items)
+      .where(isDirectChild(items.path, parent.path))
+      // use order for ordering
+      // backup order by in case two items has same ordering
+      .orderBy(() => [asc(items.order), asc(items.createdAt)]);
+  }
+
+  /**
+   * return children of given parent, ordered by order, with creator
+   * @param db
    * @param actor
    * @param parent
    * @param params
    * @returns
    */
-  async getChildren(
+  async getChildrenWithCreator(
```

Comment:
I think it was smart to split this function into the "simple" get all children and have this one where you can provide the search parameters. My question now is whether the search parameters should be optional or not. Maybe they should be required to indicate that this function does more than just return the children with the creator ? Also not sure it should be named this way. Maybe we can name it something like "getFilteredChildren" ? Or relate it to the fact that it is used to display a view in the front, so the shape is tailored to this ? These are just ideas, maybe the name is fine, it is just that on the principle I do not like to put the return type in the function name.